### PR TITLE
ACS-5659 Jakarta Migration

### PR DESCRIPTION
**Pull request by pzhyland**

### CODE REVIEW

File: repository/src/main/resources/alfresco/subsystems/Messaging/default/messaging-context.xml [hunk 1]
```diff
@@ -40,9 +39,9 @@
         <property name="configuration" ref="jmsConfig"/>
     </bean>
 
-    <bean id="activemq" class="org.apache.camel.component.activemq.ActiveMQComponent">
-        <property name="configuration" ref="jmsConfig"/>
-    </bean>
+<!--    <bean id="activemq" class="org.apache.camel.component.activemq.ActiveMQComponent">-->
+<!--        <property name="configuration" ref="jmsConfig"/>-->
+<!--    </bean>-->
```
Comment (kmagdziarz) [inline repository/src/main/resources/alfresco/subsystems/Messaging/default/messaging-context.xml:44]: Commented out.

File: remote-api/src/main/java/org/alfresco/rest/framework/core/ResourceInspector.java [hunk 2]
```diff
@@ -378,20 +383,16 @@ public static int validSuccessCode(HttpMethod httpMethod, int success)
...
-                return Status.STATUS_OK;
-        }
+        if (GET.equals(httpMethod))
+            return Status.STATUS_OK;
+        if (POST.equals(httpMethod))
+            return Status.STATUS_CREATED;
+        if (PUT.equals(httpMethod))
+            return Status.STATUS_OK;
+        if (DELETE.equals(httpMethod))
+            return Status.STATUS_NO_CONTENT;
```
Comment (kmagdziarz) [inline remote-api/src/main/java/org/alfresco/rest/framework/core/ResourceInspector.java:393]: We should use braces here to be compliant with rest of the class.

File: remote-api/src/main/java/org/alfresco/rest/framework/core/ResourceInspector.java [hunk 3]
```diff
@@ -26,6 +26,11 @@
 
 package org.alfresco.rest.framework.core;
 
+import static org.springframework.http.HttpMethod.DELETE;
+import static org.springframework.http.HttpMethod.GET;
+import static org.springframework.http.HttpMethod.POST;
+import static org.springframework.http.HttpMethod.PUT;
```
[the remark this answers is not in the document]
Comment (pzhyland) [reply remote-api/src/main/java/org/alfresco/rest/framework/core/ResourceInspector.java:32]: PMD is wrong here.

File: remote-api/src/main/java/org/alfresco/rest/framework/core/ResourceInspector.java (hunk 2 above)
Comment (pzhyland) [reply remote-api/src/main/java/org/alfresco/rest/framework/core/ResourceInspector.java]: Done

File: packaging/tests/tas-restapi/src/main/java/org/alfresco/rest/core/RestWrapper.java [hunk 4]
```diff
@@ -630,37 +631,27 @@ protected Response sendRequest(RestRequest restRequest)
...
+            returnedResponse = onRequest().body(restRequest.getBody())
+                    .post(restRequest.getPath(), restRequest.getPathParams()).andReturn();
+        else if (HttpMethod.PUT.equals(httpMethod))
+            returnedResponse = onRequest().body(restRequest.getBody())
+                    .contentType(ContentType.JSON.withCharset(restRequest.getContentType()))
+                    .put(restRequest.getPath(), restRequest.getPathParams()).andReturn();
+        else if (HttpMethod.TRACE.equals(httpMethod))
+            returnedResponse = onRequest().get(restRequest.getPath(), restRequest.getPathParams()).andReturn();
+        else
+            returnedResponse = onRequest().get(restRequest.getPath(), restRequest.getPathParams()).andReturn();
```
[the remark this answers is not in the document]
Comment (pzhyland) [reply packaging/tests/tas-restapi/src/main/java/org/alfresco/rest/core/RestWrapper.java]: Done

File: repository/src/main/resources/alfresco/subsystems/Messaging/default/messaging-context.xml (hunk 1 above)
Comment (pzhyland) [reply repository/src/main/resources/alfresco/subsystems/Messaging/default/messaging-context.xml]: Done

File: repository/pom.xml [hunk 5]
```diff
@@ -624,21 +626,21 @@
...
+<!--            <artifactId>activemq-pool</artifactId>-->
+<!--            <version>${dependency.activemq.version}</version>-->
+<!--            <exclusions>-->
+<!--                &lt;!&ndash; Duplicates classes from jakarta.transaction:jakarta.transaction-api &ndash;&gt;-->
+<!--                <exclusion>-->
+<!--                    <groupId>org.apache.geronimo.specs</groupId>-->
+<!--                    <artifactId>geronimo-jta_1.1_spec</artifactId>-->
+<!--                </exclusion>-->
+<!--            </exclusions>-->
+<!--        </dependency>-->
```
[the remark this answers is not in the document]
Comment (pzhyland) [reply repository/pom.xml]: Done

**Pull request merged**